mlir/lib/Dialect/ArmSME/Transforms/TileAllocation.cpp

```diff
@@ -488,76 +501,126 @@ coalesceTileLiveRanges(DenseMap<Value, LiveRange> &initialLiveRanges) {
   return std::move(coalescedLiveRanges);
 }
 
-/// Choose a live range to spill (via some heuristics). This picks either an
-/// active live range from `activeRanges` or the new live range `newRange`.
+/// Choose a live range to spill (via some heuristics). This picks either a live
+/// range from `activeRanges`, `inactiveRanges`, or the new live range
+/// `newRange`. Note: All live ranges in `activeRanges` and `inactiveRanges` are
+/// assumed to overlap with `newRange`.
 LiveRange *chooseSpillUsingHeuristics(ArrayRef<LiveRange *> activeRanges,
+                                      ArrayRef<LiveRange *> inactiveRanges,
```

Comment:
After re-reading this, I realised that `chooseSpillUsingHeuristics` doesn't care about the distinction into `activeRanges` and `inactiveRanges`? If that's the case, then IMO the interface should be like this:
```cpp
LiveRange *chooseSpillUsingHeuristics(ArrayRef<LiveRange *> overlappingRanges)
```

Reply:
They're passed separately to avoid allocating a new vector (when I already have a vector of active live ranges, and a vector of overlapping inactive live ranges). `llvm::concat<LiveRange>` simply creates an iterator over both of those (without allocating a new vector).

Reply:
Why couldn't we concat before calling this method? The type would be grim ...

Perhaps rename as
```cpp
LiveRange *chooseSpillUsingHeuristics(ArrayRef<LiveRange *> firstRange, LiveRange *chooseSpillUsingHeuristics(ArrayRef<LiveRange *> secondRange)
 ```
 ? And then add a comment that "this method doesn't care, but this is easier to pass as an argument"? I just find referring to "active"/"inactive" in this hook super confusing (while it shouldn't be - this method doesn't care).

Reply:
Templates to the rescue! I'll just make `chooseSpillUsingHeuristics` a template that takes an `OverlappingRangesIterator`.